### PR TITLE
Update goals.md

### PR DESCRIPTION
**Pull request by malomarrec**
This is a draft pull request with the current state of my thinking on goals for Campaigns. It is a basis for discussion, and requires work:
- align on outcomes with the team
- product research to drive understanding of the user base
- ... leading to setting the metrics that are still placeholders (<xxx>)

### CODE REVIEW

**Comment by malomarrec** [comment]
Quinn's way of putting Goal 2 it a customer meeting is "Campaigns for all". That's a nice way of outlining Campaigns adoption. Praphrasing
"Campaigns should be in the hands of every engineer, and it should be easy for others to discover campaigns because they will see Campaigns affecting their repositories in a great way.
Fixing common bad patterms is a great use case. Some fixes are so basic and repetitive that engineers should never ever have to allocate a brain cell to them once fixed with a campaign."

(Malo's note That would warrant an argument about Campaigns vs Linting, vs Static analysis, etc)

**Comment by malomarrec** [comment]
Campaigns for all - tool that every company uses or will use in the future
Open source - popular libraries can publish campaign specs to help users upgrade. Cross the public and private boundry
Invariant on your code - I always want this thing to be true
  __ Fix everything today
  __ Fix any regression entered 3 weeks from now
    __ Provide this feedback to developers as soon as possible
    __ You should get this in your editor
    __ Immediate feedback on how to make your code more compliant to standards

**Comment by malomarrec** [comment]
Campaigns is exhaustive: fix _everything_

**Comment by malomarrec** [comment]
"Enforce invariants" needs to be a big element of the vision.

File: handbook/engineering/campaigns/goals.md [hunk 1]
```diff
@@ -1,73 +1,53 @@
...
-1. Allow multiple changeset in a single repository
-1. Improve previews on updates to show the delta between existing changesets and the changesets after after the campaign spec changes are applied
-1. Allow campaign to specify target branch
-1. Create campaign spec even if subset of repos have failures
-1. Versioning/releasing of src-cli with respect to sg/sg
-1. Improved documentation of src-cli login process
-1. Add filtering/searching to campaign and changeset lists
-1. Ability to auto-merge changesets, depending on various conditions
-1. Ability to add default reviewers to changesets
+<A screenshot of ProductBoard that shows the scope of product research, engineering experiments, and development leading to those metrics and outcomes>
```
Comment (chrispine) [inline handbook/engineering/campaigns/goals.md:53]: I think we will need more than a screenshot, as this section is pulled into the [product roadmap](https://about.sourcegraph.com/handbook/product/roadmap) page.

**Review by chrispine — COMMENTED** [review]
I think this is a good start! I'm eager to see what it looks like with actual numbers.

Have you gotten any feedback on it from anyone outside of our team? As Christina is out, Nick might be a good person to reach out to.

File: handbook/engineering/campaigns/goals.md [hunk 2]
```diff
@@ -1,73 +1,53 @@
...
+## Milestones and Outcomes
 
-Currently the tools companies do have for making large scale changes are only usable by a select group of experienced developers and solve a limited number of use cases. 
+1. Campaigns can support the environment (code host, integrations, local env) of xx % of the user base
 
-### Problems
-* Campaigns does not currently allow users to open many changesets in a single repo in a way that supports monorepo workflows
-* It takes too long to merge changesets because owners are not aware the  changesets were created
-* The documentation and usability of campaigns are not sufficient to provide a frictionless experience for less experienced developers
+2. Campaigns can handle the scale of xx% of the user base
```
[the remark this answers is not in the document]
Comment (malomarrec) [reply handbook/engineering/campaigns/goals.md:24]: I would disagree with that. The call we had yesterday with a large scale customer showed that operating at scale causes a lot of tracking, UX, management of lots of PR data issues.

File: handbook/engineering/campaigns/goals.md [hunk 3]
```diff
@@ -1,73 +1,53 @@
...
 
-### Milestones
-* Robust support for monorepo workflows
-* Support adding reviewers to changesets
-* Support adding labels to changesets
-* Documentation on how to set up and start using and troubleshoot campaigns has been tested and improved
+3. Customers can create successful large scale changesets (<500+> changesets)
```
[the remark this answers is not in the document]
Comment (malomarrec) [reply handbook/engineering/campaigns/goals.md:26]: Perhpas merge 2. and 3.. What we need to shoot for however is: 
AS a campaign creator, I am confident that that I can rollout a 1,000 changeset and have them merged

**Comment by malomarrec** [comment]
Moving this discussion [here](https://docs.google.com/document/d/14KLRhuRFpmHA1QBwdidA-UPEVU8Z--joTa9xXOJV7O4/edit#), GitHub is not the easiest for long threads on multiple items

**Pull request closed without merging**